packages/hadron-document/test/document.test.ts

```diff
@@ -1558,65 +1558,373 @@ describe('Document', function () {
   });
 
   describe('array modifications', function () {
-    it('can add array elements', function () {
-      const doc = new Document({
-        a: [1, 2, 3],
-      });
-      doc.get('a')?.insertEnd(3, new Int32(4));
-      expect(doc.generateUpdateUnlessChangedInBackgroundQuery()).to.deep.equal({
-        query: { _id: null, 'a.3': { $exists: false } },
-        updateDoc: { $set: { 'a.3': new Int32(4) } },
+    context('with a plain array name', function () {
```

Comment:
These tests remain unchanged, only split into two `context()`s, one for plain names and one for dots-and-dollar names